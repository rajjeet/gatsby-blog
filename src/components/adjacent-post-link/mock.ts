export const previous = {
  fields: { slug: '/previous/' },
  frontmatter: { title: 'Introducing the New React Concurrent Mode' },
};
export const next = {
  fields: { slug: '/next/' },
  frontmatter: { title: 'The Implications of React Concurrent Mode' },
};
