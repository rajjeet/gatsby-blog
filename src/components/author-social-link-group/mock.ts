import { TProps } from './types';

export const makeProps = (): TProps => ({
  className: '',
});

