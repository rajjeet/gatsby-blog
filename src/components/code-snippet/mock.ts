import { TProps } from './types';

export const makeProps = (): TProps => ({
  children: `
const mainReducer = (state, action) => {
  console.log('Before', state);
  console.log('Action', action);
  let result = {
    count: countReducer(state.count, action)      
  };
  console.log('After', result);
  return result
};      
`,
  fileName: 'index.js',
});
