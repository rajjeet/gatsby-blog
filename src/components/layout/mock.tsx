import React from 'react';
import { TProps } from './types';
import { createMockGatsbyImageSharpFluid } from '../../utils/testing';
import { siteMetadata } from '../../../gatsby-config';

export const makeProps = (): TProps => ({
  children: <div>Sample child</div>,
  data: {
    file: createMockGatsbyImageSharpFluid.file,
    site: siteMetadata,
  },
});

