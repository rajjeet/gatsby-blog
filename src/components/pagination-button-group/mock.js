export const makeProps = () => ({
  className: {}, currentPage: 1, numOfPages: 5, paginationSlug: '',
});
