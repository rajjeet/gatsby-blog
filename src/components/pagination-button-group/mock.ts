export const makeProps = ({ currentPage = 1, numOfPages = 5 } = {}): TProps => ({
  className: '', currentPage, numOfPages, paginationSlug: '',
});

export type TProps = {
  className: string;
  currentPage: number;
  numOfPages: number;
  paginationSlug: string;
};
