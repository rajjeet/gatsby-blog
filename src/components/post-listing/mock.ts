import { TProps } from './types';
import { mockPosts } from '../../fixtures/posts';

export const makeProps = ({
  posts = mockPosts,
  heading = 'Post Listing',
  numOfPages = 5,
  currentPage = 1,
  paginationSlug = '/sample_slug/',
} = {}): TProps => ({
  posts,
  heading,
  numOfPages,
  currentPage,
  paginationSlug,
});

