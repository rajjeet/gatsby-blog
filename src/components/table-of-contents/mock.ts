import { TProps } from './types';

export const makeProps = (): TProps => ({
  items: [
    {
      url: '#introduction',
      title: 'Introduction',
    },
    {
      url: '#the-basics',
      title: 'The Basics',
      items: [
        {
          url: '#state-management-redux',
          title: 'State Management (Redux)',
        },
        {
          url: '#context-api',
          title: 'Context API',
        },
      ],
    },
    {
      url: '#manage-global-state-in-react-hooks',
      title: 'Manage Global State in React Hooks',
    },
    {
      url: '#middleware-logger',
      title: 'Middleware: Logger',
    },
  ],
});

