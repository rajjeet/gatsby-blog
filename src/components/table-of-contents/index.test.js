import React from 'react';
import { render, cleanup } from '@testing-library/react';
import TableOfContents from './index';
import { makeProps } from './mock';

afterEach(cleanup);

describe('<TableOfContents />', () => {
  it('should render', () => {
    const { asFragment } = render(<TableOfContents {...makeProps()} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('should show each link', () => {
    const { getByText } = render(<TableOfContents {...makeProps()} />);
    expect(getByText('Introduction')).toBeDefined();
    expect(getByText('The Basics')).toBeDefined();
    expect(getByText('State Management (Redux)')).toBeDefined();
    expect(getByText('Context API')).toBeDefined();
    expect(getByText('Manage Global State in React Hooks')).toBeDefined();
    expect(getByText('Middleware: Logger')).toBeDefined();
  });

  it('should show only one heading', () => {
    const { getByText } = render(<TableOfContents {...makeProps()} />);
    expect(getByText('Outline')).toBeDefined();
  });
});
