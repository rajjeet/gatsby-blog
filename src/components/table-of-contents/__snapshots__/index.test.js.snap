// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TableOfContents /> should render 1`] = `
<DocumentFragment>
  <h4>
    Outline
  </h4>
  <ul
    class="sc-bdVaJa cSwstg"
    id="static-toc"
  >
    <div>
      <a
        class="sc-bwzfXH dLeDxw"
        href="#introduction"
      >
        <li>
          Introduction
        </li>
      </a>
    </div>
    <div>
      <a
        class="sc-bwzfXH dLeDxw"
        href="#the-basics"
      >
        <li>
          The Basics
        </li>
      </a>
      <ul
        class="sc-bdVaJa cSwstg"
        id="static-toc"
      >
        <div>
          <a
            class="sc-bwzfXH dLeDxw"
            href="#state-management-redux"
          >
            <li>
              State Management (Redux)
            </li>
          </a>
        </div>
        <div>
          <a
            class="sc-bwzfXH dLeDxw"
            href="#context-api"
          >
            <li>
              Context API
            </li>
          </a>
        </div>
      </ul>
    </div>
    <div>
      <a
        class="sc-bwzfXH dLeDxw"
        href="#manage-global-state-in-react-hooks"
      >
        <li>
          Manage Global State in React Hooks
        </li>
      </a>
    </div>
    <div>
      <a
        class="sc-bwzfXH dLeDxw"
        href="#middleware-logger"
      >
        <li>
          Middleware: Logger
        </li>
      </a>
    </div>
  </ul>
</DocumentFragment>
`;
